python/cuml/tsa/arima.pyx

```diff
@@ -0,0 +1,899 @@
+
+    nvtx_range_pop()
+
+    return mu, ar, ma
+
+
+def ll_f(num_batches, nobs, order, y, x,
+         trans=True, handle=None):
+    """Computes batched loglikelihood for given parameters and series.
+
+    Parameters:
+    ----------
+    num_batches : int
+                  Number of series
+    nobs : int
+           Number of samples in each series (identical across series)
+    order : Tuple[int, int, int]
+            ARIMA Order (p, d, q)
+    y     : array like, shape = (n_samples, n_series)
+            Time series data
+    x     : array
+            dense parameter array, grouped by series,
+            and again by [(mu, ar, ma), (mu, ar, ma)]
+    trans : bool
+            Should the `jones_transform` be applied?
+            Note: The parameters from a `fit()` model are already transformed.
+    handle : cumlHandle (optional)
+             The cumlHandle to be used.
+    """
+
+    cdef vector[double] vec_loglike
+
+    vec_loglike = _batched_loglike(num_batches,
+                                   nobs, order,
+                                   y, x,
+                                   trans, handle)
+
+    loglike = np.zeros(num_batches)
+    for i in range(num_batches):
+        loglike[i] = vec_loglike[i]
+
+    return loglike
+
+
+def ll_gf(num_batches, nobs, num_parameters, order,
+          y, x, h=1e-8, trans=True, handle=None):
+    """Computes gradient (via finite differencing) of the batched
+    loglikelihood.
+
+    Parameters:
+    ----------
+    num_batches : int
+                  Number of series
+    nobs : int
+           Number of samples in each series (identical across series)
+    num_parameters : int
+                     The number of parameters per series (p + d + q)
+    order : Tuple[int, int, int]
+            ARIMA Order (p, d, q)
+    y     : array like, shape = (n_samples, n_series)
+            Time series data
+    x     : array
+            dense parameter array, grouped by series, and
+            again by [(mu, ar, ma), (mu, ar, ma)]
+    h     : float
+            The finite-difference stepsize
+    trans : bool
+            Should the `jones_transform` be applied?
+            Note: The parameters from a `fit()` model are already transformed.
+    handle : cumlHandle (optional)
+             The cumlHandle to be used.
+
+    """
+    nvtx_range_push("ll_gf")
+
+    fd = np.zeros(num_parameters)
+
+    grad = np.zeros(len(x))
+
+    assert (len(x) / num_parameters) == float(num_batches)
+    for i in range(num_parameters):
+        fd[i] = h
+
+        # duplicate the perturbation across batches (they are independent)
+        fdph = np.tile(fd, num_batches)
+
+        # reset perturbation
+        fd[i] = 0.0
+
+        ll_b_ph = ll_f(num_batches, nobs, order, y, x+fdph,
+                       trans=trans, handle=handle)
+        ll_b_mh = ll_f(num_batches, nobs, order, y, x-fdph,
+                       trans=trans, handle=handle)
+
+        np.seterr(all='raise')
```

Comment:
This line has side effects on other components using `numpy`. E.g `umap`, which causes CI Failures.
(thanks @dantegd for finding that!)

Reply:
Solved